### ISSUE TITLE
Features obtained with shapelet transform

### ISSUE THREAD

**Issue by ricardovvargas** [blue-yonder/tsfresh#448]
Hi. Do you guys know the shapelet transform? It can be used to extract features that can be included in set of features extracted by tsfresh.

The tslearn package has functions [to learn shaplets](https://tslearn.readthedocs.io/en/latest/gen_modules/shapelets/ShapeletModel/tslearn.shapelets.ShapeletModel.fit.html#tslearn.shapelets.ShapeletModel.fit) and [to do the transformation](https://tslearn.readthedocs.io/en/latest/gen_modules/shapelets/ShapeletModel/tslearn.shapelets.ShapeletModel.transform.html#tslearn.shapelets.ShapeletModel.transform). 

This transformation is documented in these papers: [here](https://www.ismll.uni-hildesheim.de/pub/pdfs/grabocka2014e-kdd.pdf) and [here](http://wan.poly.edu/KDD2012/docs/p289.pdf).

It is kind of expensive to extract this type of feature but I would be happy to check if it improves or not the performance of classifiers.

**Comment by GillesVandewiele** [comment]
Hey @ricardovvargas , the shapelet transformation does not really fit in the workflow of tsfresh, since it is a supervised approach (it needs labeled timseries to calculate how good certain shapelet candidates are). And as you mentioned, it can be quite expensive to calculate them. We had an entire talk about it in #111 

That said, you can still extract shapelets with whatever package you like and join the distances of those shapelets to your samples with the features extracted by tsfresh. Just split your data in train and test set before extracting the shapelets.

**Comment by ricardovvargas** [comment]
Hi @GillesVandewiele. So you guys have already discussed about shapelet. Nice! However in #111 it is NOT discussed about shapelet TRANSFORM. @MaxBenChrist has said that "We have the convention that features must be real valued numbers". 

As we know a shapelet is a subsequence but a shapelet transformation result is a real valued number that can be considered a feature which can be very helpful in some domains. 

You said that "shapelet transformation does not really fit in the workflow of tsfresh, since it is a supervised approach" but I have noticed that tsfresh can be used together with I sklearn classifiers in a supervised approach to select best features.

Yes, we can extract shapelets, do a shapelet transformation and join its features with features extracted by tsfresh but my suggestion was/is integrate this operation because tsfresh proposes automatically calculate a large number of time series characteristics and not everybody knows features extracted by shapelet transformation. Maybe tslearn package can offer all functions needed. Yes, this kind of feature can be expensive but it can be included just in the ComprehensiveFCParameters or other group of features. The best part is that tsfresh users would have more features to choose.

Anyway, this is just a suggestion. Tsfresh is a very nice package. Congratulations.

**Comment by GillesVandewiele** [comment]
2 posts under the one your citing is more relevant:

By this you get a feature calculator with a state. Essentially you have to train the calculator (to get the shapelets) and then predict with it ( to get the distance to the class).

`This means that tsfresh would have need to have separated data sets for train / test set. Up until now, the users can extract features for their whole data and do the train/test split by themselves.`

**Comment by nils-braun** [comment]
Seems this issue is stalled and/or solved. Closing now.
